measures/views.py

```diff
@@ -1116,48 +1116,47 @@ class MeasuresCreateProcessQueue(
         "common.change_trackedmodel",
     ]
     template_name = "measures/create-process-queue.jinja"
-    filter_by_fields = ["all", "processing", "completed", "failed"]
-    sort_by_fields = [
-        "workbasket_id",
-        "submitted_date",
-        "submitted_by",
-        "status",
-        "object_count",
-        "action",
-    ]
-
-    def get_queryset(self):
-        return MeasuresBulkCreator.objects.filter(
-            workbasket__status=WorkflowStatus.EDITING,
-        ).order_by("-created_at")
+    model = MeasuresBulkCreator
+    queryset = MeasuresBulkCreator.objects.filter(
+        workbasket__status=WorkflowStatus.EDITING,
+    ).order_by("-created_at")
+    filterset_class = MeasureCreateTaskFilter
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
+        context["selected_link"] = "all"
         processing_state = self.request.GET.get("processing_state")
-        context["selected_filter"] = "all"
-        if processing_state in [
-            ProcessingState.CURRENTLY_PROCESSING,
-            ProcessingState.AWAITING_PROCESSING,
-        ]:
-            context["selected_filter"] = "processing"
+
+        if processing_state == "PROCESSING":
+            context["selected_link"] = "processing"
         elif processing_state == ProcessingState.CANCELLED:
-            context["selected_filter"] = "cancelled"
+            context["selected_link"] = "cancelled"
         elif processing_state == ProcessingState.FAILED_PROCESSING:
-            context["selected_filter"] = "failed"
+            context["selected_link"] = "failed"
         elif processing_state == ProcessingState.SUCCESSFULLY_PROCESSED:
-            context["selected_filter"] = "completed"
-
+            context["selected_link"] = "completed"
         # Provide template access to some UI / view utility functions.
         context["status_tag_generator"] = self.status_tag_generator
         context["can_cancel_task"] = self.can_cancel_task
         context["is_task_failed"] = self.is_task_failed
-
         # Apply the TAP standard date format within the UI.
         context["datetime_format"] = settings.DATETIME_FORMAT
-
+        if context["selected_link"] == "processing":
+            context["object_list"] = self.get_processing_queryset()
         return context
 
+    def get_processing_queryset(self):
+        """Returns a combined queryset of tasks either AWAITING_PROCESSING or
+        CURRENTLY_PROCESSING."""
+        awaiting_processing = self.queryset.filter(
+            processing_state="AWAITING_PROCESSING",
+        )
+        currently_processing = self.queryset.filter(
+            processing_state="CURRENTLY_PROCESSING",
+        )
+        return awaiting_processing.union(currently_processing)
```

Comment:
I don't think a union is the right expression here. A pair or or'd Q objects are the right way to go. Also, accessing state values via `ProcessingState` is a safer and more maintainable approach:
```suggestion
        return self.queryset.filter(
            Q(processing_state=ProcessingState.AWAITING_PROCESSING)
            | Q(processing_state=ProcessingState.CURRENTLY_PROCESSING)
        )
```